java-client/session/src/main/java/io/deephaven/client/impl/SharedId.java

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.client.impl;
+
+import com.google.protobuf.ByteString;
+import io.deephaven.proto.util.SharedTicketHelper;
+
+import java.util.Objects;
+
+/**
+ * An opaque holder for a shared object ID.
+ */
+public class SharedId implements HasTicketId, HasPathId {
+
+    private final ByteString sharedId;
+
+    public SharedId(final ByteString sharedId) {
```

Comment:
I'm _not_ sure if we've had `com.google.protobuf` as part of the public API. I wonder if `byte[]` would be more appropriate? (similar to `io.deephaven.client.impl.TicketId`)